source/DataObjectsLoader.h

```diff
@@ -0,0 +1,53 @@
+/* DataObjectsLoader.h
+Copyright (c) 2022 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef DATA_OBJECTS_LOADER_H_
+#define DATA_OBJECTS_LOADER_H_
+
+#include "InterfaceObjects.h"
+#include "UniverseObjects.h"
+
+#include <future>
+
+
+
+// This class contains the loader for serialized game data. The serialized
+// game data contains various types of intermixed data (interfaces, outfits,
+// ships, missions) so the loader does need to load to different Object stores.
+class DataObjectsLoader{
```

Comment:
I don't think it makes sense for this class to be a normal object. I think it should be a static class whose only (public) function is `Load`. And if that's the case we might as well put the Load function back into GameData. What do you think?

Reply:
I was thinking in a different direction of making the DataObjectsLoader even more generic in the future and instantiating it whenever we need to load some data (and destroying it when we are done loading such data);
- Do we load game-data-files? Then we instantiate a DataObjectsLoader with UniverseObjects and InterfaceObjects as "to be loaded items".
- Do we load a save-game? Then we instantiate a DataObjectsLoader with a PlayerInfo object as "to be loaded item".
- Does ES-launcher need to load something? Then it could re-use the loader as well (outside of the rest of the game context, [as suggested by tehhowh](https://github.com/endless-sky/endless-sky/pull/6502#discussion_r857107457)).

But the code for this load function is not very big, so I also agree that there are good reasons to keep this code as a static function and just implement a specific version for every load scenario. (Non-asynchronous loading is safer for most applications anyway. ;-) )

Reply:
Note that the code from the Load function before this PR was located in UniverseObjects (as non-static method), not in GameData. Moving code towards a static function doesn't help to increase test-ability of the code.